httpbakery/browser.go

```diff
@@ -26,16 +32,116 @@ func OpenWebBrowser(url *url.URL) error {
 	return err
 }
 
-// WebBrowserVisitor holds an interactor that supports the "Interactive"
-// method by opening a web browser at the required location.
-var WebBrowserVisitor Visitor = webBrowserVisitor{}
+type WebBrowserInteractor struct {
+	// OpenWebBrowser is used to visit a page in
+	// the user's web browser. If it's nil, the
+	// OpenWebBrowser function will be used.
+	OpenWebBrowser func(*url.URL) error
+}
+
+func (WebBrowserInteractor) Kind() string {
+	return WebBrowserInteractionKind
+}
+
+// SetInteraction sets interaction information on the given error,
+// which should be an interaction-required error that's about
+// to be returned from a discharge request.
+//
+// The visitURL parameter holds a URL that should be
+// visited by the user in a web browser; the waitTokenURL parameter
+// holds a URL that can be long-polled to acquire the resulting
+// discharge token.
+//
+// The waitURL parameter is for legacy clients that don't
+// understand about discharge tokens and should
+// hold a URL that can be long-polled to acquire the
+// discharge macaroon.
+func (wi WebBrowserInteractor) SetInteraction(e *Error, visitURL, waitTokenURL, waitURL string) {
```

Comment:
This method seems backwards. If I'm a server that wants to allow the WebBrowser interaction, why would I want to create one just to add the URLs? Why not just make this a function it doesn't seem to use the receiver at all.